### PR TITLE
ci: add more versions of numpy/scipy/pypy

### PR DESCRIPTION
**Pull request by henryiii**
Signed-off-by: Henry Schreiner <henryschreineriii@gmail.com>

## Description

Adding some extra testing. Hoping to manage with only binary.

## Suggested changelog entry:

```rst
* Test on PyPy 3.10
```

### CODE REVIEW

File: tests/requirements.txt [hunk 1]
```diff
@@ -1,9 +1,11 @@
-build==0.8.0
-numpy==1.21.5; platform_python_implementation=="PyPy" and sys_platform=="linux" and python_version=="3.7"
-numpy==1.19.3; platform_python_implementation!="PyPy" and python_version=="3.6"
-numpy==1.21.5; platform_python_implementation!="PyPy" and python_version>="3.7" and python_version<"3.10"
-numpy==1.22.2; platform_python_implementation!="PyPy" and python_version>="3.10" and python_version<"3.11"
-pytest==7.0.0
+--only-binary=:all:
+build~=0.10.0
```
Comment (henryiii) [inline tests/requirements.txt:2]:
```suggestion
build~=0.9.0
```

File: tests/requirements.txt [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
-numpy==1.19.3; platform_python_implementation!="PyPy" and python_version=="3.6"
-numpy==1.21.5; platform_python_implementation!="PyPy" and python_version>="3.7" and python_version<"3.10"
-numpy==1.22.2; platform_python_implementation!="PyPy" and python_version>="3.10" and python_version<"3.11"
-pytest==7.0.0; platform_python_implementation!="PyPy" and python_version=="3.6"
-pytest==7.2.0; platform_python_implementation!="PyPy" and python_version>="3.7"
+--only-binary=:all:
+build~=0.9; python_version=="3.6"
+build~=1.0; python_version>="3.7"
+numpy~=1.20.0; python_version=="3.7" and platform_python_implementation=="PyPy"
+numpy~=1.22.2; python_version=="3.8" and platform_python_implementation=="PyPy"
```
Comment (henryiii) [inline tests/requirements.txt:5]:
```suggestion
numpy~=1.23.0; python_version=="3.8" and platform_python_implementation=="PyPy"
```

**Review by rwgk — APPROVED** [review]
Just curious: is `~=` a new feature?

**Comment by henryiii** [comment]
No, it’s been around since the beginning. It means let the final digit get larger. So ~=1.2 would be >=1.2,<2 and ~=1.2.0 would be >=1.2,<1.3.

**Pull request merged**